sdk/eventhub/Azure.Messaging.EventHubs/CONTRIBUTING.md

```diff
@@ -27,9 +25,118 @@ Tests in the Event Hubs client library are split into two categories:
+
+## Development environment setup
+
+### Prerequisites
+
+- **Azure Subscription:**  To use Azure services, including Azure Event Hubs, you'll need a subscription.  If you do not have an existing Azure account, you may sign up for a [free trial](https://azure.microsoft.com/free/dotnet/) or use your [Visual Studio Subscription](https://visualstudio.microsoft.com/subscriptions/) benefits when you [create an account](https://account.windowsazure.com/Home/Index).
+
+- **PowerShell:** To use the Azure SDK automation for development tools and setting up your testing environment, PowerShell Core version 7.0 or greater is needed.  This version is available cross-platform and can be installed from the [PowerShell Core repository](https://github.com/PowerShell/PowerShell/blob/master/README.md).
+
+- **Azure PowerShell Module:** For PowerShell to interact with Azure, the [Azure Az module]((https://docs.microsoft.com/en-us/powershell/azure/install-az-ps)) is needed.  
+
+### Creating Azure resources for tests
+
+#### Automating creation with `NewTestResources.ps1`
+
+To setup your environment to run the Live tests, the recommended approach is to use the Azure SDK [Test Resources](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/README.md) tooling to deploy the Event Hubs test resources and create the Azure service principal needed by the tests.  
+
+The following commands will do so, using the `BaseName` as a prefix when naming the service principal and other resources.  The principal created will be granted the "Owner" role for the new resource group but will have no access to other resources in the subscription. 
+
+```powershell
+Connect-AzAccount -Subscription '<< AZURE SUBSCRIPTION ID >>'
+
+<repository-root>/eng/common/TestResources/New-TestResources.ps1 `
+    -BaseName '<< MEMORABLE VALUE (example: azsdk) >>' `
+    -ServiceDirectory 'eventhub' `
+    -SubscriptionId '<< AZURE SUBSCRIPTION ID >>' `
+    -ResourceGroupName '<< NAME FOR RESOURCE GROUP >>' `
+    -Location '<< AZURE REGION CODE (example: eastus) >>' `
+    -ArmTemplateParameters @{ perTestExecutionLimitMinutes = '5' }
+```
+
+The `ResourceGroupName` and `Location` parameters are optional; if not provided, the `BaseName` will be used to generate the name of the resource group and will be located in `westus2`.  For a list of the Azure locations valid for Event Hubs, the following command can be used:
+
+```powershell
+Get-AzLocation `
+    | Where { $_.Providers.Contains("Microsoft.EventHub") } `
+    | Select DisplayName, Location `
+    | Format-Table
+```
+
+If you prefer to use an existing service principal with the Live tests, the following set of parameters will instruct the script to create a new resource group for the tests and assign the "Owner" role for the resource group to the specified `TestAppliaticationId`.
```

Comment:
Assuming we need the actual objectId rather than the App Id:

```suggestion
If you prefer to use an existing service principal with the Live tests, the following set of parameters will instruct the script to create a new resource group for the tests and assign the "Owner" role for the resource group to the specified service principal. To specify a service principal, specify it's Object Id in the `TestAppliaticationId` parameter.
```

Also here is an az command to get it given the App Id

`az ad sp show --id <appId> --query objectId`

Reply:
Nope; the grant uses the App Id.    Assuming the provisioner has rights to do so, the script will also query the OID for the principal automatically, so it's cool just to specify the App Id and secret.  _(the exception here is if you're using a service account to provision and haven't granted it extra API rights to Graph)_

```powershell
New-AzRoleAssignment -RoleDefinitionName "Owner" -ApplicationId "$TestApplicationId" -ResourceGroupName "$ResourceGroupName"
```